test/built-ins/Set/prototype/difference/converts-negative-zero.js

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2023 Anthony Frehner and Kevin Gibbons. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-set.prototype.difference
+description: Set.prototype.difference converts -0𝔽 to +0𝔽
+info: |
+    7.b.ii If nextValue is -0𝔽, set nextValue to +0𝔽.
+features: [set-methods]
+includes: [compareArray.js]
+---*/
+
+const setlikeWithMinusZero = {
+  size: 1,
+  has: function () {
+    throw new Test262Error("Set.prototype.difference should not call its argument's has method when this.size > arg.size");
```

Comment:
I would expect a polyfill to check its receiver's `.size` (or rather to invoke the intrinsic `Set.prototype.size` getter on its receiver) and then branch as in step 6 of `Set.prototype.difference`, so I would in fact expect a polyfill to avoid this line.

Reply:
Right, but one of the branches seems to require access to the internal data, which a polyfill can’t get.

Reply:
I'm confused. Both branches in this method, and indeed all of the branches in all of the methods, require reading the `[[SetData]]` from the receiver. Obviously polyfills can't do that directly, but they can use the set prototype intrinsics. That's what I'd expect them to do. There is nothing particularly special about this branch in that regard.

Can you say in more detail exactly why you think this branch requires access to internal slots in a way the other branch does not?

Reply:
yep, you're right, it was just that the optimization I'd been using wasn't correct. Sorry for the noise!